src/main/java/org/mule/service/http/impl/service/client/HttpMessageLogger.java

```diff
@@ -31,17 +34,24 @@ public HttpMessageLogger(final LoggerType loggerType) {
 
   @Override
   public void onDataReceivedEvent(Connection connection, Buffer buffer) {
-    logBuffer(buffer);
+    logBuffer(buffer, getLogContext(connection));
   }
 
   @Override
   public void onDataSentEvent(Connection connection, Buffer buffer) {
-    logBuffer(buffer);
+    logBuffer(buffer, getLogContext(connection));
   }
 
-  private void logBuffer(Buffer buffer) {
+  private static Map<String, String> getLogContext(Connection connection) {
+    Map<String, String> logContext = (Map<String, String>) connection.getAttributes().getAttribute("logContext");
```

Comment:
Yes, connections are shared. We should verify whether it's the same instance at this point.